pkg/nodeconfig/nodeconfig.go

```diff
@@ -460,9 +427,9 @@ func (nc *nodeConfig) setNode(quickCheck bool) error {
 	return errors.Wrapf(err, "unable to find node with address %s", instanceAddress)
 }
 
-// waitForNodeAnnotation checks if the node object has the given annotation and waits for retry.Interval seconds and
-// returns an error if the annotation does not appear in that time frame.
-func (nc *nodeConfig) waitForNodeAnnotation(annotation string) error {
+// waitForNodeAnnotation checks if the node object has the given annotation with the expected value.
+// Waits for retry.Interval seconds and returns an error if it does not appear in that time frame.
+func (nc *nodeConfig) waitForNodeAnnotation(annotation, expectedVal string) error {
```

Comment:
This function can be moved to the metadata package and be made specific to the version annotation.